### DISCUSSION TITLE
keyrings

### DISCUSSION THREAD

**Discussion by Heaven1** [DarkCastleMUD/DarkCastle#119] [Q&A] [answered]
so i was monkeying around earlier and coded up keyrings.

**Comment by jhhudso** [comment, marked as answer]
I like the idea and will implement it using your pull request patch as a base correcting as needed. Sorry I didn't respond sooner but I only today noticed it. I'll respond sooner in the future.
In the future please format the code a bit more so it's easier to read on github's pull-requests web page. Lately I use the free Visual Studio Code editor and its C++ formatting on Linux. 
Ensure all strings sent to users end with \r\n. This is standard for MUDs as well as SMTP, HTTP and other Internet protocols. This way text will look correct for Linux, Windows and OSX users.
Please test your code before submitting. If you need more files added to the repository in order to test, I'll commit them.
Since DIKUMUD changed their license to LGPL, DC coders agreed to license DarkCastle under LGPL. If you have a problem with your code being licensed under that let me know and I'll remove it.
As for assigning ITEM_KEYRING as 1U<<14 to be used as a more_flags bitvector, I think it would be better as a item type_flag such as number 34 then you increase ITEM_TYPE_MAX to 34. This is defined earlier in obj.h. I'll make that change.
I'm not sure I like the idea of keys being auto-sac'd when duplicates are added to the keyring. Maybe some players will want to save duplicates of some keys so they can give an important duplicate key to a friend.
Please email me at the address shown for julian in the EMAIL helpfile in Dark Castle or contact me on Dark Castle so we can discuss this code or future code changes you submit.

**Reply by jhhudso** [reply]
Also normally I'd suggest create an issue instead of a discussion to discuss changes, bugs, etc. Then submit PRs linked to the issue.
It appeared like you submitted 2 PRs then closed them then opened this 3rd PR.

**Reply by bdunavant** [reply]
Just a note, we forked off of DIKU a long long time ago.   Just because the current DIKU is LGPL, that doesn't mean we are.   It's a question I'd never really thought about since we have never been for-profit.   If you want to figure out the correct license, we'd have to figure out when we actually forked off DIKU and if they even had a license on it back then.

**Reply by jhhudso** [reply]
I disagree because the owners of the DikuMUD said their code can be licensed under either license now. They said derivatives are not automatically LGPL now, it's up to every derivative to decide. Our MUD is a derivative of DIKU so it can be licensed under either depending on what the coders who's contributed to DC decide.

**Reply by jhhudso** [reply]
Other MUD code bases that are derivatives of DikuMUD have said the same thing now due to the notice from DikuMUD. For example, CircleMUD, which was a pretty famous, old codebase that's derivative of DikuMUD, like DC, now say they too are LGPL due to the notice that DikuMUD has given everyone who is derived from them the freedom to be LGPL vs DikuMUD's old restrictive non-commercial only license we use to all have to stick with but no longer have to.

**Reply by jhhudso** [reply]
Are you saying that you personally would prefer any code you wrote that was a derivative of DikuMUD to be licensed under the older license and not the newer license?

**Reply by jhhudso** [reply]
https://dikumud.com/dikumud-license/
https://www.circlemud.org/license.html

**Comment by bdunavant** [comment]
Closing the loop here, I had forgotten that we (several, but not all, DC coders), decided to change DC to be under LGPL as well.   We should likely add a LICENSE file to the root of the code, that reflects this.
